src/nvim/os/env.c

```diff
@@ -412,6 +412,15 @@ void os_get_hostname(char *hostname, size_t size)
 static char *homedir = NULL;
 static char *os_homedir(void);
 
+const char *os_get_homedir(void)
```

Comment:
Add a function about the difference of this function with `os_homedir()`?

Reply:
Or maybe one of these functions should be renamed, as it's confusing to have both `os_homedir()` and `os_get_homedir()`.

Reply:
You mean describe the differences in the commit message?

Reply:
Sorry, I mean add a comment above this function.

Reply:
does it make sense to rename the existing `os_homedir` to `uv_homedir` (and keep it private/static), then name this new function `os_homedir` ? And the docstring on the new function can mention:

    /// Gets the "real", resolved homedir. See comment on `homedir`.

Reply:
`uv_` prefix is for libuv functions. This is a wrapper around a libuv function, so I think it should at least be `os_uv_homedir`.

Reply:
This good?

Reply:
I think you should reference the comment above the `homedir` variable.
> ```
> /// Gets the "real", resolved user home directory. See comment on `homedir`.
> ```